.upstream-tests/test/std/utilities/optional/optional.object/optional.object.observe/dereference.pass.cpp

```diff
@@ -0,0 +1,71 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11
+// <cuda/std/optional>
+
+// constexpr T& optional<T>::operator*() &;
+
+#include <cuda/std/optional>
+#include <cuda/std/type_traits>
+#include <cuda/std/cassert>
+
+#include "test_macros.h"
+
+using cuda::std::optional;
+
+struct X
+{
+    __host__ __device__
+    constexpr int test() const& {return 3;}
+    __host__ __device__
+    int test() & {return 4;}
+    __host__ __device__
+    constexpr int test() const&& {return 5;}
+    __host__ __device__
+    int test() && {return 6;}
+};
+
+struct Y
+{
+    __host__ __device__
+    constexpr int test() {return 7;}
+};
+
+__host__ __device__
+constexpr int
+test()
+{
+    optional<Y> opt{Y{}};
+    return (*opt).test();
+}
+
+int main(int, char**)
+{
+    {
+        [[maybe_unused]] optional<X> opt;
+        ASSERT_SAME_TYPE(decltype(*opt), X&);
+        LIBCPP_STATIC_ASSERT(noexcept(*opt), "");
```

Comment:
I do not think we should use that macro

Reply:
Why not? I mean it's probably unnecessary here, because it's only our code that's tested by this test and so this extension is always present, and this is a macro for testing libc++ extensions, but I really don't see the need to generate churn of removing its usage.